opensrp-chw-core/src/test/java/org/smartregister/chw/core/fragment/CoreBirthNotificationFragmentTest.java

```diff
@@ -0,0 +1,328 @@
+
+    @Mock
+    private CoreBirthNotificationFragment coreFpRegisterFragment;
+
+    @Mock
+    private CommonRepository commonRepository;
+
+    @Mock
+    private View view;
+
+    private FragmentActivity activity;
+    private ActivityController<AppCompatActivity> activityController;
+
+    @Mock
+    private ImageView imageView;
+
+    @Mock
+    private ProgressBar clientsProgressView;
+
+    @Mock
+    private TextView textView;
+
+    @Mock
+    private RelativeLayout relativeLayout;
+
+    @Mock
+    private EditText editText;
+
+    @Mock
+    private TextWatcher textWatcher;
+
+    @Mock
+    private View.OnKeyListener hideKeyboard;
+
+    @Mock
+    private ProgressBar syncProgressBar;
+
+    @Mock
+    private CoreChildRegisterFragmentContract.Presenter presenter;
+
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        coreFpRegisterFragment = Mockito.mock(CoreBirthNotificationFragment.class, Mockito.CALLS_REAL_METHODS);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "presenter", presenter);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "view", view);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "dueOnlyLayout", view);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "clientsProgressView", clientsProgressView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "dueFilterActive", true);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "qrCodeScanImageView", imageView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "headerTextDisplay", textView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "filterStatus", textView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "filterRelativeLayout", relativeLayout);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "searchView", editText);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "textWatcher", textWatcher);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "hideKeyboard", hideKeyboard);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "syncProgressBar", syncProgressBar);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "syncButton", imageView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "globalQrSearch", false);
+
+        CoreLibrary.init(context);
+        when(context.commonrepository(anyString())).thenReturn(commonRepository);
+        activityController = Robolectric.buildActivity(AppCompatActivity.class).create().resume();
+        activity = activityController.get();
+        Context.bindtypes = new ArrayList<>();
+        Whitebox.setInternalState(coreFpRegisterFragment, "clientsView", clientsView);
+        Whitebox.setInternalState(coreFpRegisterFragment, "presenter", presenter);
+        SyncStatusBroadcastReceiver.init(activity);
+    }
+
+    @Test
+    public void testSetupViews() {
```

Comment:
Kindly follow the test naming convention `test[MethodName][ExpectedResult][Parameters]`